### PR TITLE
feat(machine): add machine cloud instance watcher

### PR DESCRIPTION
**Pull request by nvinuesa**
This patch adds a simple watcher for the machine cloud instance table
that will be recruited in the uniter worker.

## Checklist

- [X] Code style: imports ordered, good names, simple structure, etc
- [X] Comments saying why design decisions were made
- [X] Go unit tests, with comments saying what you're testing
- [ ] ~[Integration tests](https://github.com/juju/juju/tree/main/tests), with comments saying what you're testing~
- [ ] ~[doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages~

## QA steps

This watcher is not wired-up yet, so unit tests should pass:

```
TEST_PACKAGES="./domain/machine/... -count=1 -race -check.vv" make run-go-tests
```

## Links

**Jira card:** JUJU-6284

### CODE REVIEW

**Review by manadart — APPROVED** [review]
Thanks.

File: domain/schema/model/sql/0002-changelog-namespaces.sql [hunk 1]
```diff
@@ -18,4 +18,5 @@ INSERT INTO change_log_namespace VALUES
 (14, 'secret_reference', 'Secret reference changes based on UUID'),
 (15, 'subnet', 'Subnet changes based on UUID'),
 (16, 'machine', 'Machine changes based on UUID'),
-(17, 'user_public_ssh_key', 'User public ssh key changes based on id');
+(17, 'machine_cloud_instance', 'Machine cloud instance changes base on machine UUID'),
```
[the remark this answers is not in the document]
Comment (nvinuesa) [reply domain/schema/model/sql/0002-changelog-namespaces.sql]: Fixed.

**Pull request merged**